### PR TITLE
ID-8

### PR DESCRIPTION
**Pull request by xiangzhe0812**
Get an ETH private and public key pair once user enter mnemonic

How to test 
1. running backend server 
2. using post man to get (http://localhost:3001/api/keypair) and give request body like 
   {
    "mnemonicKey":"tornado **********"
   }
expect get result  like 
{
    "address": "dfgsdfgdfs",
    "privateKey": "sdfgdfgdfg",
    "publicKey": "dfgdfgdfg"
}

### CODE REVIEW

**Comment by bitcoinbrisbane** [comment]
@xiangzhe0812 i think we have our wires crossed ID-8 was supposed to be on the mobile device.  A users keys should only be on their device.

File: app/helpers/setting/EthKeypair.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { Wallet } from "ethers";
+
+export const GetEthKeypair = async (mnemonicKey: string) => {
+  const walletMnemonic = Wallet.fromMnemonic(mnemonicKey);
+  const walletPrivateKey = new Wallet(walletMnemonic.privateKey);
+
+  if (walletMnemonic.address !== walletPrivateKey.address) {
+  }
```
Comment (ahinchliff) [inline app/helpers/setting/EthKeypair.ts:9]: this if block is empty

File: app/navigation/SettingsStackNavigator.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    <Stack.Navigator>
+      <Stack.Screen
+        name="Settings List"
+        component={Settings}
+        options={({ navigation }) => ({
+          headerLeft: () => <HeaderLeft navigation={navigation} />,
+          headerTitle: () => <HeaderTitle title="Settings" />,
+        })}
+      />
+      <Stack.Screen
```
Comment (ahinchliff) [inline app/navigation/SettingsStackNavigator.tsx:21]: Arent each of these screens exactly the same?

File: app/screens/setting/index.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  const rootStore = useRootStore();
+  const settings = rootStore.Assets.selectedSetting;
+
+  if (!settings) {
+    return <View>{/* TODO: error handling for this case */}</View>;
+  }
+
+  const renderSetting = (key: string) => {
+    switch (key) {
+      case "1x00":
```
Comment (ahinchliff) [inline app/screens/setting/index.tsx:17]: why do the settings need a key? I still havent quite got my head around what the key is for. Is the the key the identiy from DID? @bitcoinbrisbane

File: app/screens/settings/index.tsx [hunk 4]
```diff
@@ -1,36 +1,71 @@
...
-const Settings = () => {
+const SettingSelector = () => {
   const rootStore = useRootStore();
-  const assetStore = rootStore.Assets;
-  return(
-    <View style={styles.root}>
+  const navigation = useNavigation();
+  const settings = rootStore.Assets.settings;
+
+  console.log(settings);
```
Comment (ahinchliff) [inline app/screens/settings/index.tsx]: console.log

File: app/screens/settings/index.tsx [hunk 5]
```diff
@@ -1,36 +1,71 @@
...
+  useEffect(() => {
+    rootStore.Assets.loadSettings();
```
Comment (ahinchliff) [inline app/screens/settings/index.tsx:37]: I dont understand why we are loading the settings? What advantage is there?

**Review by ahinchliff — CHANGES_REQUESTED** [review]
Few small changes requested and I have a few questions

**Review by bitcoinbrisbane — APPROVED** [review]
LGTM

**Review by rellfy — APPROVED** [review]

**Review by ahinchliff — APPROVED** [review]

**Pull request merged**